source/Nevermore/Advanced/IConnectionAndTransaction.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nevermore.Advanced
+{
+    public interface IConnectionAndTransaction : IDisposable
```

Comment:
Nevermore doesn't have a single implementation of `IAsyncDisposable` right now. I decided to leave it out of scope but it could be done as a separate PR for sure.

My implementation of this does implement `IAsyncDisposable`, though it only plays a role if you maintain ownership over it, it doesn't get called if you give Nevermore ownership then dispose the Nevermore transaction.